src/tools/tidy/src/deps.rs

```diff
@@ -29,47 +33,425 @@ static LICENSES: &'static [&'static str] = &[
 // tooling. It is _crucial_ that no exception crates be dependencies
 // of the Rust runtime (std / test).
 static EXCEPTIONS: &'static [&'static str] = &[
-    "mdbook", // MPL2, mdbook
-    "openssl", // BSD+advertising clause, cargo, mdbook
-    "pest", // MPL2, mdbook via handlebars
-    "thread-id", // Apache-2.0, mdbook
-    "toml-query", // MPL-2.0, mdbook
-    "is-match", // MPL-2.0, mdbook
-    "cssparser", // MPL-2.0, rustdoc
-    "smallvec", // MPL-2.0, rustdoc
+    "mdbook",             // MPL2, mdbook
+    "openssl",            // BSD+advertising clause, cargo, mdbook
+    "pest",               // MPL2, mdbook via handlebars
+    "thread-id",          // Apache-2.0, mdbook
+    "toml-query",         // MPL-2.0, mdbook
+    "is-match",           // MPL-2.0, mdbook
+    "cssparser",          // MPL-2.0, rustdoc
+    "smallvec",           // MPL-2.0, rustdoc
     "fuchsia-zircon-sys", // BSD-3-Clause, rustdoc, rustc, cargo
-    "fuchsia-zircon", // BSD-3-Clause, rustdoc, rustc, cargo (jobserver & tempdir)
-    "cssparser-macros", // MPL-2.0, rustdoc
-    "selectors", // MPL-2.0, rustdoc
-    "clippy_lints", // MPL-2.0 rls
+    "fuchsia-zircon",     // BSD-3-Clause, rustdoc, rustc, cargo (jobserver & tempdir)
+    "cssparser-macros",   // MPL-2.0, rustdoc
+    "selectors",          // MPL-2.0, rustdoc
+    "clippy_lints",       // MPL-2.0 rls
+];
+
+// Whitelist of crates rustc is allowed to depend on. Avoid adding to the list if possible.
+static WHITELIST: &'static [(&'static str, &'static str)] = &[
+//    ("advapi32-sys", "0.2.0"),
```

Comment:
I think it's probably ok to only check the name of the dependency and not worry about the version for now, upgrading across major versions and such tends to not bring in *too* large of a change at least!

Reply:
I argue that it's important for security purposes to whitelist exact versions and to vet those versions before before upgrading in any way.

Reply:
Oh sure yeah, but I think we'll want to consider this further down the road. If we were to lock down all the versions of everything it'd cause I think picking up deps from crates.io to be too painful (or otherwise routine updates)

Reply:
I think this is part of the downside of having a super-awesome dependency system: it becomes easy to have too many dependencies. Argue that that makes code quality, security, and build times all worse.

Reply:
Sure but I think it's also best to do this incrementally rather than all at once, can this switch to just verifying the names through a whitelist?

Reply:
I can if you would like, but I would really like not to. I feel like adding a bit of friction to adding and updating dependencies would be healthy; in this case, you would need to update `WHITELIST`. We can later have an RFC to potentially add more restrictions to PRs that change `WHITELIST`... I see that as an incremental approach as well (we add a mechanism, then add policy through an RFC).